pkg/controller/composite/controller.go

```diff
@@ -359,6 +366,20 @@ func (pc *parentController) updateParentObject(old, cur interface{}) {
 	// different status (e.g. you have some incrementing counter).
 	// Doing that is an anti-pattern anyway because status generation should be
 	// idempotent if nothing meaningful has actually changed in the system.
+	if parentOld, ok := old.(*unstructured.Unstructured); ok {
```

Comment:
1. flip the condition, first check if  `ignoreStatusChange` is enabled, then do the comparison